internal/pkg/earpointer/analysis.go

```diff
@@ -271,15 +275,40 @@ func (vis *visitor) visitFieldAddr(addr *ssa.FieldAddr) {
 }
 
 func (vis *visitor) visitField(field *ssa.Field) {
-	// "t1 = t0.x" is implemented by t0[x -> t1].
+	// "t1 = t0.x" is implemented by t0[x -> addr], addr --> t1,
+	// where addr is the reference to the field address.
 	obj := field.X
 	fd := makeField(obj.Type(), field.Field)
-	vis.processHeapAccess(field, obj, *fd)
+	state := vis.state
+	for _, c := range vis.getContexts(field) {
+		objv := MakeReference(c, obj)
+		fmap := state.PartitionFieldMap(state.representative(objv))
+		fref := MakeReference(c, field)
+		// If obj's partition already has a mapping for this field, unify fref
+		// with the suitable partition. Otherwise set fref as the ref.
+		if v, ok := fmap[*fd]; ok {
+			// Implement "fmap[fd -> addr], addr --> fref".
+			state.Unify(vis.getPointee(v), fref)
+		} else {
+			fmap[*fd] = fref // over-approximation
+		}
+	}
 }
 
 func (vis *visitor) visitIndexAddr(addr *ssa.IndexAddr) {
```

Comment:
Adding the "constant" field is to provide some extra information about the field, e.g. 

m["a"] = r0
m["b"] = r1
r2 = m["a"]
r3 = m[x]

we know that r2 can only be r0, but r3 can be {r0, r1}. Of course, the current revision actually over-approximates r2 to be {r0, r1} as well since  m["a"] = r0 results in m[anyIndex -> {r0}]. However, it is good to keep the constant field information as we can refine this in the future. For example, in a must-may analysis, we know that r2 may be {r0, r1}, but r2 must be {r0}.